chapters/cls-linear.qmd

```diff
@@ -11,41 +11,259 @@ knitr:
+
+Also, the annual minimum temperature, dew temperature, January minimum temperature, and maximum vapor show bimodality in the distributions. The year of inspection is also interesting; the data collection was sparse before 2011, and subsequent years contain a few hundred data points per year before beginning to drop off in 2021. These characteristics are not indicative of problems with data quality, but it can be important to know that they exist when debugging why a model is underperforming or showing odd results.
+
+::: {#tbl-forested-numeric}
+
+```{r}
+#| label: forested-numeric
+#| echo: false
+
+vert_forest |>
+  summarize(
+    Minimum = min(value),
+    Mean = mean(value),
+    Max = max(value),
+    `Std. Dev` = sd(value),
+    Skewness = e1071::skewness(value),
+    Distribution = list(value),
+    .by = c(Predictor)
+  ) |>
+  arrange(Predictor) |>
+  gt() |>
+  fmt_number(columns = c(-Distribution), n_sigfig = 3) |>
+  gt_plt_dist(
+    Distribution,
+    type = "histogram",
+    same_limit = FALSE,
+    line_color = "#222",
+    fill_color = "white"
+  ) |>
+  cols_width(Predictor ~ pct(25))
+
+```
+
+Histograms and statistical summaries of the numeric predictors in the Washington State training set. 
+
+:::
+
+
+Also, recall that @fig-corr-plot previously described the correlation structure of these predictors. There were several clusters of predictors with strong magnitudes of correlation. This implies that there is some redundancy of information in these features. However, machine learning is often “a game of inches,” where even redundant predictors can contribute incremental improvements in performance. In any case, the analyses in this chapter will be profoundly affected by this characteristic; it will be discussed in more detail below. 
+
+Individually, how does each of these features appear to relate to the outcome? To assess this, we binned each predictor into roughly 20 groups based on percentiles and used these groups (each containing about `r floor(mean(percentiles$total))` locations) to compute the rate of forestation and 90% confidence intervals^[Binning is used here as a visualization tool; we re-emphasize that converting numeric predictors into categorical features is problematic.]. As an exploratory tool, we can use these binned versions of the data to see potential relationships with the outcome. @fig-forest-percentiles shows the profiles. Note that there is enough data in each bin to make the confidence intervals very close to the estimated rates. 
```

Comment:
re:

>  Note that there is enough data in each bin to make the confidence intervals very close to the estimated rates. 

It's surprising to me that the intervals are so tight around `year` even when there's so much variation in the mean in the upper 25%?


<img width="237" height="126" alt="Screenshot 2025-07-21 at 3 01 41 PM" src="https://github.com/user-attachments/assets/058f92ad-87aa-4c73-b773-12cf469d2294" />

Seeing that binned variation, I would have guessed that the intervals would be wider.

Reply:
I felt the same way and initially worried that I used `conf.level = 0.1` instead of `conf.level = 0.9` but it is correct. I tend to use 90% intervals (more narrow than 95%) so that is part of it. 

There are about 230 rows per point so the +/- is about 4%